webroot/js/utils/user-colors.js

```diff
@@ -8,7 +8,7 @@ export function messageBubbleColorForString(str) {
   // Tweak these to adjust the result of the color
   const saturation = 25;
   const lightness = 45;
-  const alpha = 0.3;
+  const alpha = '--var(--message-background-alpha)';
```

Comment:
You don't need the `--` in front of the `var`, else it'll break the alpha calculation and turn them all transparent.

Reply:
Sorry, didn’t know what I was thinking :) (I can’t really test on localhost as I haven’t had time to figure out how to stream to it to start the chat).

Reply:
Oh!  A little tip if for testing, there's a little script in `test` you can use to stream yourself test content using any random video you have around:

`./test/ocTestStream.sh ~/Videos/big_buck_bunny.mp4`

And it'll just stream to localhost using the default stream key.  But you could add your own rtmp destination as an argument if you wanted to send a test elsewhere.

Reply:
Ah, that’s very useful, thank you. Although, in retrospect, I could’ve just used ngrok to expose my local server and streamed to it from my mixer too :)

Reply:
@gabek I just tried this with a vanilla from source install (all default settings) and got the following error. Any ideas? :)

```
❯ ./ocTestStream.sh '/home/aral/Downloads/Big_Buck_Bunny_1080_10s_1MB.mp4'

Streaming a loop of 1 videos to rtmp://localhost/live/abc123.  Warning: If these files differ greatly in formats transitioning from one to another may not always work correctly...  ctl+c to exit
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x55e850a37da0] Auto-inserting h264_mp4toannexb bitstream filter
Input #0, concat, from 'list.txt':
  Duration: N/A, start: 0.000000, bitrate: 812 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], 812 kb/s, 30 fps, 30 tbr, 15360 tbn, 60 tbc
    Metadata:
      handler_name    : VideoHandler
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> h264 (libx264))
Press [q] to stop, [?] for help
[libx264 @ 0x55e850a3e7c0] using SAR=1/1
[libx264 @ 0x55e850a3e7c0] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
[libx264 @ 0x55e850a3e7c0] profile Main, level 4.0
[libx264 @ 0x55e850a3e7c0] 264 - core 152 r2854 e9a5903 - H.264/MPEG-4 AVC codec - Copyleft 2003-2017 - http://www.videolan.org/x264.html - options: cabac=1 ref=1 deblock=1:0:0 analyse=0x1:0x111 me=hex subme=2 psy=1 psy_rd=1.00:0.00 mixed_ref=0 me_range=16 chroma_me=1 trellis=0 8x8dct=0 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=0 threads=18 lookahead_threads=6 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=1 keyint=250 keyint_min=25 scenecut=0 intra_refresh=0 rc_lookahead=10 rc=abr mbtree=1 bitrate=1300 ratetol=1.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
Output #0, flv, to 'rtmp://localhost/live/abc123':
  Metadata:
    encoder         : Lavf57.83.100
    Stream #0:0(und): Video: h264 (libx264) ([7][0][0][0] / 0x0007), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], q=-1--1, 1300 kb/s, 30 fps, 1k tbn, 30 tbc
    Metadata:
      handler_name    : VideoHandler
      encoder         : Lavc57.107.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/1300000 buffer size: 0 vbv_delay: -1
av_interleaved_write_frame(): Broken pipe time=00:00:00.00 bitrate=N/A speed=   0x    
[flv @ 0x55e850a7bb80] Failed to update header with correct duration.
[flv @ 0x55e850a7bb80] Failed to update header with correct filesize.
Error writing trailer of rtmp://localhost/live/abc123: Broken pipe
frame=   33 fps= 20 q=40.0 Lsize=     107kB time=-00:00:00.06 bitrate=N/A speed=N/A    
video:107kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.287435%
[libx264 @ 0x55e850a3e7c0] frame I:1     Avg QP:31.71  size:108538
[libx264 @ 0x55e850a3e7c0] frame P:8     Avg QP:39.37  size:  3004
[libx264 @ 0x55e850a3e7c0] frame B:24    Avg QP:41.98  size:   252
[libx264 @ 0x55e850a3e7c0] consecutive B-frames:  3.0%  0.0%  0.0% 97.0%
[libx264 @ 0x55e850a3e7c0] mb I  I16..4: 32.4%  0.0% 67.6%
[libx264 @ 0x55e850a3e7c0] mb P  I16..4:  1.7%  0.0%  0.0%  P16..4: 12.5%  2.2%  0.9%  0.0%  0.0%    skip:82.6%
[libx264 @ 0x55e850a3e7c0] mb B  I16..4:  0.0%  0.0%  0.0%  B16..8:  1.0%  0.1%  0.0%  direct: 0.1%  skip:98.8%  L0:45.9% L1:52.8% BI: 1.3%
[libx264 @ 0x55e850a3e7c0] final ratefactor: 30.33
[libx264 @ 0x55e850a3e7c0] coded y,uvDC,uvAC intra: 67.2% 55.3% 17.6% inter: 0.1% 0.3% 0.0%
[libx264 @ 0x55e850a3e7c0] i16 v,h,dc,p: 24% 24% 33% 19%
[libx264 @ 0x55e850a3e7c0] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 18% 14% 18% 10% 11%  9%  8%  8%  6%
[libx264 @ 0x55e850a3e7c0] i8c dc,h,v,p: 54% 25% 13%  7%
[libx264 @ 0x55e850a3e7c0] Weighted P-Frames: Y:0.0% UV:0.0%
[libx264 @ 0x55e850a3e7c0] kb/s:1008.20
Conversion failed!
```

Reply:
The only thing that stands out is that Copyleft string of 2017.  I'm not sure if that specifically matters, but compared to mine it's a bit older:
`[libx264 @ 0xcad1480] 264 - core 161 r3018 db0d417 - H.264/MPEG-4 AVC codec - Copyleft 2003-2020`
So I guess make sure the version of ffmpeg you're using is somewhat recent.

Are there any errors on the Owncast side?